### PR TITLE
Add CI pipelines

### PR DESCRIPTION
**Pull request by SoorajModi**
**Why**

- Most open source projects have CI, we should add it to our project to better emulate FOSS

---

**What**

- Added CI for the whole repo
- Added CI for the `hello_world` example submission

---

**Verification**

- If you go to the actions tab, you should see these pipelines running and succeeding

---

**Checklist:**

- [X] This contribution abides by all rules

### CODE REVIEW

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+name: CI
```
Comment (MarshallAsch) [inline .github/workflows/CI.yml:1]: Missing the training new lines at the end of some files

File: .github/workflows/hello-world-ci.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
+name: hello-world-ci
+
+on: [push, pull_request]
```
Comment (MarshallAsch) [inline .github/workflows/hello-world-ci.yml]: This should also use the `on.paths` block to only run this if the files in that directory were touched. The majority of the time this pipeline will not actually do anything meaningful.

File: .github/workflows/hello-world-ci.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+jobs:
+  build:
```
Comment (MarshallAsch) [inline .github/workflows/hello-world-ci.yml]: This job should probably be called something different like `run_hello` to more accurately reflect what it does.

File: .github/workflows/hello-world-ci.yml [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: ./submissions/hello_world
+    strategy:
+      matrix:
+        python-version: [3.8]
+
+    steps:
+      - uses: actions/checkout@v2
```
Comment (MarshallAsch) [inline .github/workflows/hello-world-ci.yml]: Use the newer checkup action (the other workflow uses v3)

**Review by MarshallAsch — CHANGES_REQUESTED** [review]

File: .github/workflows/hello-world-ci.yml (hunk 4 above)
Comment (SoorajModi) [reply .github/workflows/hello-world-ci.yml]: Updated checkout to `v3` and setup-python to `v4`

File: .github/workflows/CI.yml (hunk 1 above)
Comment (SoorajModi) [reply .github/workflows/CI.yml:1]: Added

File: .github/workflows/hello-world-ci.yml (hunk 2 above)
Comment (SoorajModi) [reply .github/workflows/hello-world-ci.yml]: Good call, updated push, but I still wanna run all CI pipelines on new PRs

File: .github/workflows/hello-world-ci.yml (hunk 3 above)
Comment (SoorajModi) [reply .github/workflows/hello-world-ci.yml]: I like the name currently, `hello-world-ci` reflects the name of the project and that its performing CI actions, while `run_hello` is more accurate, I like how general it the current name is

Comment (MarshallAsch) [reply .github/workflows/hello-world-ci.yml]: The current name of the job is build, (the comment now seems to be on a different line)

Comment (SoorajModi) [reply .github/workflows/hello-world-ci.yml]: Updated name to `run_hello_world`

**Review by MarshallAsch — APPROVED** [review]

**Pull request merged**